release-calendar/src/client/components/Calendar.tsx

```diff
@@ -62,15 +62,20 @@ export class Calendar extends React.Component<CalendarProps, CalendarState> {
   }
 
   async componentDidUpdate(prevProps: CalendarProps): Promise<void> {
-    if (prevProps.singleRelease != this.props.singleRelease) {
-      if (this.props.singleRelease != null) {
-        const release = await this.apiService.getRelease(
-          this.props.singleRelease,
-        );
-        this.setState({singleEvents: getSingleReleaseEvents(release)});
-      } else {
-        this.setState({singleEvents: []});
-      }
+    if (
+      this.props.singleRelease != prevProps.singleRelease &&
+      this.props.singleRelease != this.state.singleEvents.releaseName &&
+      this.props.singleRelease != null
+    ) {
+      const release = await this.apiService.getRelease(
+        this.props.singleRelease,
+      );
+      this.setState({
+        singleEvents: {
+          releaseName: this.props.singleRelease,
+          events: getSingleReleaseEvents(release, this.state.allEvents),
+        },
+      });
```

Comment:
I don't think we should be saving `singleEvents` when a user clicks on another filter because this muddles the separation of `calendarProps`, which has filter criteria, and `state`, which has events. We should keep these separate.

Reply:
I'm happy to change it back but can you explain this more? what you mean when you say filter criteria

Reply:
By filter criteria I mean - what channels are selected, if any? which release is selected, if any?

Because `state` in this PR includes a selected release, it's possible that it does not match `calendarProps`, which could lead to bugs in the future. (A developer could easily assume that the calendar should show whatever is in `state`, but in this case it could be stale). We should let `calendarProps` drive the content of `state`, not the other way around.

Reply:
oh, I see, that makes sense.